File: python/anagram/anagram.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def detect_anagrams(word, anagrams):
    anagram_list = []
    for a in anagrams:
        if word == a:
            # same word we break
            break
        # Lowercase, sort string and join without spaces
        elif ''.join(sorted(word.lower())) == ''.join(sorted(a.lower())):
            anagram_list.append(a)

    return anagram_list

File: javascript/anagram/anagram.js
var anagram = function(str) {
	// sort letters in a string (lowercase)
	var sorted = function(w) {
		return w.toLowerCase().split("").sort().join('');
	};

	this.word = str;
	this.matches = function(words) {
		// make an array if arguments are string(s)
		if (typeof words === 'string') {
			words = [].slice.apply(arguments);
		}

		var sortedWord = sorted(this.word);
		var sameWord = [];

		for (var i = 0; i < words.length; i++) {
			// same exact word, case insensitively? Discard
			if (this.word.toLowerCase() === words[i].toLowerCase()) {
				break
			} else if (sortedWord === sorted(words[i])) {
					sameWord.push(words[i]);
			}
		}
		return sameWord;
	}
	return this;
};

module.exports = anagram;
